web_programming/fetch_anime_and_play.py

```diff
@@ -0,0 +1,194 @@
+                            " ", ""
+                        ),
+                        "url": children.find("a")["href"],
+                    }
+                )
+        except (KeyError, NotFoundErr):
+            pass
+
+    return episode_list
+
+
+def get_anime_episode(episode_endpoint: str) -> list:
+
+    """[summary]
+
+    Get click url and download url from episode url
+
+    >>> type(get_anime_episode("/watch/kimetsu-no-yaiba/1"))
+    <class 'list'>
+
+    Args:
+        episode_endpoint (str): [Endpoint of episode]
+
+    Raises:
+        e: [description]
+
+    Returns:
+        [list]: [List of download and watch url]
+    """
+
+    episode_page_url = f"{BASE_URL}{episode_endpoint}"
+
+    response = requests.get(
+        url=episode_page_url, headers={"User-Agent": UserAgent().chrome}
+    )
+    response.raise_for_status()
+
+    soup = BeautifulSoup(response.text, "html.parser")
+
+    try:
+        episode_url = soup.find("iframe", {"id": "playerframe"})["src"]
+        download_url = episode_url.replace("/embed/", "/playlist/") + ".m3u8"
+    except (KeyError, NotFoundErr) as e:
+        raise e
+
+    return [f"{BASE_URL}{episode_url}", f"{BASE_URL}{download_url}"]
+
+
+if __name__ == "__main__":
+
+    try:
+
+        anime_name = input("Enter anime name: ").strip()
+        anime_list = search_scraper(anime_name)
+        print("\n")
+
+        if len(anime_list) == 0:
+            print("No anime found with this name")
+        else:
+
+            print(f"Found {len(anime_list)} results: ")
+            for (i, anime) in enumerate(anime_list):
+                anime_title = anime["title"]
+                print(f"{i+1}. {anime_title}")
+
+            anime_choice = int(
+                input("\nPlease choose from the following list: ").strip()
+            )
+            chosen_anime = anime_list[anime_choice - 1]
+            print(f"You chose {chosen_anime['title']}. Searching for episodes...")
+
+            episode_list = search_anime_episode_list(chosen_anime["url"])
+            if len(episode_list) == 0:
+                print("No episode found for this anime")
+            else:
+                print(f"Found {len(episode_list)} results: ")
+                for (i, episode) in enumerate(episode_list):
+                    print(f"{i+1}. {episode['title']}")
+
+                episode_choice = int(
+                    input("\nChoose an episode by serial no: ").strip()
+                )
+                chosen_episode = episode_list[episode_choice - 1]
+                print(f"You chose {chosen_episode['title']}. Searching...")
+
+                episode_url, download_url = get_anime_episode(chosen_episode["url"])
+                print(f"\nTo watch, ctrl+click on {episode_url}.")
+                print(f"To download, ctrl+click on {download_url}.")
+
+    except (ValueError, IndexError, TypeError) as e:
+        raise e
```

Comment:
We can lose the try / except if we are not going to handle the errors.

Reply:
removed the block.